src/Common/src/System/Net/SafeCloseSocket.Unix.cs

```diff
@@ -161,6 +161,13 @@ public int SendTimeout
             }
         }
 
+        internal bool IsDisconnected { get; private set; } = false;
+
+        internal void SetToDisconnected()
+        {
+            IsDisconnected = true;
+        }
```

Comment:
Socket already has an _isDisconnected field; can we just check that rather than adding another field (the implicit one backing IsDisconnected above) to track the same state?  Or is there something different/special about this one?